pkg/kpipelines/syntax/pipeline.go

```diff
@@ -674,49 +730,16 @@ func stageToTask(s Stage, pipelineIdentifier string, buildIdentifier string, nam
 		// We don't want to dupe volumes for the Task if there are multiple steps
 		volumes := make(map[string]corev1.Volume)
 		for _, step := range s.Steps {
-			// TODO: Ignoring everything but commands right now, but will eventually need to handle syntactic sugar steps too
-			if step.Command != "" {
-				stepImage := agent.Image
-				if !equality.Semantic.DeepEqual(step.Agent, Agent{}) {
-					stepImage = step.Agent.Image
-				}
-
-				var c corev1.Container
-
-				if podTemplates != nil && podTemplates[stepImage] != nil {
-					podTemplate := podTemplates[stepImage]
-					containers := podTemplate.Spec.Containers
-					for _, volume := range podTemplate.Spec.Volumes {
-						volumes[volume.Name] = volume
-					}
-					c = containers[0]
-					c.Args = append([]string{step.Command}, step.Arguments...)
-				} else {
-					c = corev1.Container{
-						Image:   stepImage,
-						Command: []string{step.Command},
-						Args:    step.Arguments,
-					}
-				}
-				stepCounter++
-				c.Name = "step" + strconv.Itoa(1+stepCounter)
-
-				c.Stdin = false
-				c.TTY = false
-
-				c.Env = env
+			actualSteps, stepVolumes, newCounter, err := generateSteps(step, agent.Image, env, podTemplates, stepCounter)
+			if err != nil {
+				return nil, err
+			}
 
-				t.Spec.Steps = append(t.Spec.Steps, c)
+			stepCounter = newCounter
 
-				/*				t.Spec.Steps = append(t.Spec.Steps, corev1.Container{
-								Name:    MangleToRfc1035Label(fmt.Sprintf("stage-%s-step-%d", s.Name, i), suffix),
-								Env:     env,
-								Image:   stepImage,
-								Command: []string{step.Command},
-								Args:    step.Arguments,
-							})*/
-			} else {
-				return nil, errors.New("syntactic sugar steps not yet supported")
+			t.Spec.Steps = append(t.Spec.Steps, actualSteps...)
+			for k, v := range stepVolumes {
+				volumes[k] = v
```

Comment:
Just noticed that we append volumes looping over a map on Line 746, so that will be nondeterministic, but we don't have any tests for volumes IIRC so not a problem yet.

Reply:
d'oh!

Reply:
And fwiw I introduced the bug in abayer/jx#13

Reply:
heh, well, I'll clean that up in a followup.